solution/index.js

```diff
@@ -0,0 +1,545 @@
+        }
+        // Reverts the bin to its originial state.
+        const deleteAllButtonChildren = deleteAllButton.querySelectorAll("*");
+        for(let child of deleteAllButtonChildren){
+            child.remove();
+        }
+    }
+}
+
+// Handles a drag and drop of a task.
+function dragNDrop(event){
+    const task=event.target;
+    task.draggable = true;
+    const parentListIndex = task.parentElement.getAttribute("data-list")
+    const parentListArray = listAsArray(parentListIndex);
+    const draggables = document.querySelectorAll(".sections > ul > li");
+    task.addEventListener("dragstart", dragStart);
+    
+    let dragStartIndex;
+    function dragEnd(){
+        task.removeEventListener("dragstart", dragStart);
+    }
+    function dragOver(event){
+        event.preventDefault();
+    }
+    function dragEnter(){
+        this.classList.add("over");
+    }
+    function dragLeave(){
+        this.classList.remove("over");
+    }
+    function dragStart(){
+        draggables.forEach(draggable => {
+            draggable.addEventListener("dragover", dragOver);
+            draggable.addEventListener("drop", dragDrop);
+            draggable.addEventListener("dragenter", dragEnter);
+            draggable.addEventListener("dragleave", dragLeave);
+        })
+        task.addEventListener("dragend", dragEnd);
+        dragStartIndex = this.closest("li").getAttribute("data-task-index");
+    }
+    function dragDrop(){
+        const dragEndIndex = this.getAttribute("data-task-index");
+        swapItems(dragStartIndex, dragEndIndex);
+
+        this.classList.remove("over");
+    }
+    // Swaps the task with another task when dropped on one.
+    function swapItems(fromIndex, toIndex){
+        const itemOne = parentListArray[fromIndex];
+        const itemTwo = parentListArray[toIndex];
+        editLocalStorage(parentListIndex, fromIndex, itemTwo.innerText)
+        editLocalStorage(parentListIndex, toIndex, itemOne.innerText)
+
+        draggables.forEach(draggable => {
+            draggable.removeEventListener("dragover", dragOver);
+            draggable.removeEventListener("drop", dragDrop);
+            draggable.removeEventListener("dragenter", dragEnter);
+            draggable.removeEventListener("dragleave", dragLeave);
+        })
+        task.removeEventListener("dragstart", dragStart);
+        clearExistingTasks();
+        generateTasks();
+    }
+    task.addEventListener("mouseleave" , ()=>{
+        task.removeEventListener("dragstart", dragStart);
+    })
+}
+
+// Background for the page.
+document.body.style.backgroundImage = "url('background.jpeg')";
+document.body.style.backgroundSize = "100vw 100vh";
+
+// First time setup.
+generateTasks();
+addIndexToLists();
+addIndexToTasks();
+
+const taskSections = document.querySelector("#task-sections");
+taskSections.addEventListener("click", handleSubmitClick);
+taskSections.addEventListener("mouseover", mouseOverList);
```

Comment:
You could put this listener directly on all the task elements, and you'd have immediate access to them in the handler.